src/test/java/com/ge/predix/log4j1/PredixLayoutTest.java

```diff
@@ -276,13 +279,124 @@ public void testPredixEncoderWithNonStringMessageAndMessageLineSeparator() {
         Assert.assertEquals(expected, actual);
     }
 
-    private LoggingEvent createLogEvent(final Object message) {
+    @Test
+    public void testPredixLayoutWithCustomCorrelationKey() {
+
+        HashMap<String, String> mdc = getMDC();
+        mdc.remove(CORRELATION_KEY);
+        mdc.put(CORRELATION_KEY_OTHER, CORRELATION_VALUE_OTHER);
+
+        LoggingEvent input = createLogEvent(getMsg(), mdc);
+
+        String expected = "{\"time\":\"" + ISO_DATE_FORMAT.format(new Date(input.getTimeStamp())) + "\",\"tnt\":\""
+                          + ZONE_VALUE + "\",\"corr\":\"" + CORRELATION_VALUE_OTHER + "\",\"appn\":\"" + APP_NAME_VALUE
+                          + "\",\"dpmt\":\"" + APP_ID_VALUE + "\",\"inst\":\"" + INSTANCE_ID_VALUE + "\",\"tid\":\"" + THREAD_NAME
+                          + "\",\"mod\":\"" + FILE_NAME + "\",\"lvl\":\"" + Level.INFO.toString()
+                          + "\",\"msg\":{\"width\":4,\"length\":3,\"units\":\"inches\",\"height\":5}}\n";
+
+        PredixLayout predixLayout = new PredixLayout();
+        predixLayout.setMessageLineSeparatorRegex(null);
+        predixLayout.setCorrelationKey(CORRELATION_KEY_OTHER);
+        String actual = predixLayout.format(input);
+        System.out.println(actual);
```

Comment:
NIT: Guess we can remove this line

Reply:
It's harmless and makes it easier to debug failed tests.